app/controllers/concerns/decidim/vocdoni/has_vote_flow.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Decidim
+  module Vocdoni
+    # Common logic for the vote flow
+    module HasVoteFlow
+      extend ActiveSupport::Concern
+
+      def vocdoni_api_endpoint_env
+        Decidim::Vocdoni.config.api_endpoint_env
+      end
+
+      def preview_mode?
+        return @preview_mode if defined?(@preview_mode)
+
+        @preview_mode = !election.published? || !election.started?
```

Comment:
```suggestion
        @preview_mode ||= !election.published? || !election.started?
```
Can't we just do this?

Reply:
Yes, nice catch, much cleaner!

This was actually copied from `decidim-elections`: https://github.com/decidim/decidim/blob/84b97f904c72ba2bb9037721e63b4fb3b29fbaa0/decidim-elections/app/controllers/concerns/decidim/elections/has_vote_flow.rb#L26